File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.py
def solution(n):
    result = ''
    while n > 0:
        a, b = divmod(n, 3)
        if b == 0:
            result = '4' + result
            n = a - 1
        else:
            result = str(b) + result
            n = a
    return result

File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.js
function solution(n) {
    let result = '';
    while(n > 0){
        let m = n % 3;
        if(m === 0){
            result = '4' + result
            n = Math.floor(n / 3) - 1;
        } else {
            result = m + result;
            n = Math.floor( n / 3);
        }
    }
    return result
}